### DISCUSSION TITLE
DirectoryScanJob DIRECTORY_PROVIDER_NAME

### DISCUSSION THREAD

**Discussion by billjwolff** [quartznet/quartznet#2201] [Q&A] [answered]
I am trying to get a DirectoryScanJob working in .NET Core Minimal API. The job fails without a DIRECTORY_PROVIDER_NAME. Where can I find valid provider values?

**Comment by lahma** [comment, marked as answer]
You can find the source code for 3.x version here: https://github.com/quartznet/quartznet/tree/3.x/src/Quartz.Jobs/Job

**Comment by billjwolff** [comment]
That helps, I set both DIRECTORY_NAMES and DIRECTORY_PROVIDER_NAME to null. Now the directory scan job is triggered. Unfortunately, it still throws this error: Couldn't update job data: Unable to serialize instance of 'System.IO.FileInfo'. This seems specific to .NET Core which I am using. Is there some workaround for Newtonsoft.Json trying to serialize Fileinfo?

**Reply by lahma** [reply]
Might need code changes to get rid of FileInfo usage and keep it backwards compatible. A PR would be appreciated.
